go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/jaegertracing/jaeger
 
-go 1.21
+go 1.21.0
```

Comment:
according to LLM specifying patch version in this directive is not recommended. The directive controls source code compatibility, which is not affected by the patch version.

On that topic, most of our CI workflows are using `go-version: 1.22.x`. It's fine for test-only workflows, but others are used to build final artifacts, and you could argue not fixing a precise patch version means the builds are not repeatable. That sounds like the issue the security scanner should be flagging, not the go.mod directive

Reply:
For reference, all official examples of `go` directive use major.minor version -- https://go.dev/ref/mod#go-mod-file-go